Explorer/Assets/DCL/InWorldCamera/PhotoDetail/VisiblePersonController.cs

```diff
@@ -0,0 +1,204 @@
+        private readonly PhotoDetailPoolManager photoDetailPoolManager;
+        private readonly ChatEntryConfigurationSO chatEntryConfiguration;
+
+        private VisiblePerson? visiblePerson;
+        private bool isShowingWearables;
+        private bool wearablesLoaded;
+        private CancellationTokenSource loadWearablesCts = new();
+
+        public VisiblePersonController(VisiblePersonView view,
+            IWebRequestController webRequestController,
+            IProfileRepository profileRepository,
+            IMVCManager mvcManager,
+            IWearableStorage wearableStorage,
+            IWearablesProvider wearablesProvider,
+            PhotoDetailPoolManager photoDetailPoolManager,
+            ChatEntryConfigurationSO chatEntryConfiguration)
+        {
+            this.view = view;
+            this.profileRepository = profileRepository;
+            this.mvcManager = mvcManager;
+            this.wearableStorage = wearableStorage;
+            this.wearablesProvider = wearablesProvider;
+            this.photoDetailPoolManager = photoDetailPoolManager;
+            this.chatEntryConfiguration = chatEntryConfiguration;
+
+            this.imageController = new ImageController(view.profileImage, webRequestController);
+            this.view.userProfileButton.onClick.AddListener(ShowPersonPassportClicked);
+            this.view.expandWearableButton.onClick.AddListener(WearableListButtonClicked);
+        }
+
+        public async UniTask SetupAsync(VisiblePerson visiblePerson, CancellationToken ct)
+        {
+            this.visiblePerson = visiblePerson;
+            isShowingWearables = false;
+            wearablesLoaded = false;
+            view.expandWearableButtonImage.localScale = Vector3.one;
+            view.wearableListContainer.gameObject.SetActive(false);
+            view.wearableListLoadingSpinner.SetActive(false);
+            view.wearableListEmptyMessage.SetActive(false);
+            loadWearablesCts = loadWearablesCts.SafeRestart();
+            Color userColor = chatEntryConfiguration.GetNameColor(visiblePerson.userName);
+
+            view.userName.text = visiblePerson.userName;
+            view.userName.color = userColor;
+            view.userNameTag.text = $"#{visiblePerson.userAddress[^4..]}";
+
+            Profile? profile = await profileRepository.GetAsync(visiblePerson.userAddress, ct);
+            if (profile is not null)
+                view.userNameTag.gameObject.SetActive(!profile.HasClaimedName);
+            else
+                view.userNameTag.gameObject.SetActive(false);
+
+            view.faceFrame.color = userColor;
+            userColor.r += 0.3f;
+            userColor.g += 0.3f;
+            userColor.b += 0.3f;
+            view.faceRim.color = userColor;
+
+            //Check: ProfileWidgetController.cs @ line 68
+            // await imageController!.RequestImageAsync(profile.Avatar.FaceSnapshotUrl, ct);
+        }
```

Comment:
as stated in the comment, it is bound to [this](https://github.com/decentraland/unity-explorer/blob/707fd6db601bbcd8f585d7e10ad1b4135fdd9630/Explorer/Assets/DCL/UI/Profile/ProfileElements/ProfileWidgetController.cs#L69) so that as soon as that is solved, this can be solved as well